src/presign/participant.rs

```diff
@@ -643,6 +644,7 @@ impl PresignParticipant {
         // Check consistency across all Gamma values
         for r3_pub in r3_pubs.iter() {
             if r3_pub.Gamma != r3_private.Gamma {
+                error!("Mismatch in Gamma values for r3_pub and r3_private");
```

Comment:
thought: this is ok for now, but I think I'd like to look into this more at some point. Could this be caused by parties using mismatched presign records? I think the `r3_pub`s are input from other players, so I'm not quite sure that this is actually an internal invariant problem.

Reply:
I guess it depends on whether Gamma is checked in the `verify()` call in Presign round three's `Public::from_message()`? That function does use Gamma in something, but I'd have to do more digging to see if it's in a way that also binds Gamma to a certain value. Can probably just call it ProtocolError for now